extensions/hibernate-search-standalone-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/standalone/elasticsearch/runtime/graal/Substitute_JandexBehavior.java

```diff
@@ -0,0 +1,20 @@
+package io.quarkus.hibernate.search.standalone.elasticsearch.runtime.graal;
+
+import org.hibernate.search.util.common.jar.spi.JandexBehavior;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+/**
+ * Disallow the use of Jandex so that relevant code can be DCEd
+ * (otherwise native compilation would fail as Jandex is not available at runtime).
+ */
+@TargetClass(className = "org.hibernate.search.util.common.jar.spi.JandexBehavior")
```

Comment:
This will work if you have both extensions? I'm not sure how GraalVM handles it?

Reply:
I'm not sure.

A fix would require introducing a third JAR that both extensions would depend on, though... Perhaps a `quarkus-hibernate-search-mapper-pojo-base` extension (deployment + runtime, to regroup some more common code in it).

Would that additional extension be fine with you?

Reply:
I would test it and see how it goes. If it doesn't break, then I think we can live with it. If it does break, I think we can live with it for this first iteration and we can solve it later.

Reply:
I just tried and it won't work anyway because `@SearchEntity` gets picked up by the `hibernate-search-orm` extension, which will choke on those entity types since they're not ORM entity types.

I'll add this as a limitation and create an issue to handle this, and the native compilation if necessary.

Reply:
Created #39517